homeassistant/components/econet/sensor.py

```diff
@@ -111,3 +134,13 @@ def unique_id(self) -> str:
         return (
             f"{self._econet.device_id}_{self._econet.device_name}_{self._device_name}"
         )
+    
+    @property
+    def state_class(self):
+        '''Return the state class'''
+        return SENSOR_NAMES_TO_STATE_CLASS[self._device_name]
+    
+    @property
+    def device_class(self):
+        '''Return the device class'''
+        return SENSOR_NAMES_TO_DEVICE_CLASS[self._device_name]
```

Comment:
I suggest to use attributes instead, and set these inside `__init__` method:
```python
    def __init__(self, econet_device, device_name):
        """Initialize."""
        super().__init__(econet_device)
        self._econet = econet_device
        self._device_name = device_name
        self._attr_state_class = SENSOR_NAMES_TO_STATE_CLASS[self._device_name]
        self._attr_device_class = SENSOR_NAMES_TO_DEVICE_CLASS[self._device_name]
```

The same could probably be applied to `name` and `unique_id` with only `native_value` kept as a property.

Reply:
Or, even better, adjust this integration to use `SensorEntityDescription`s instead.

Reply:
> Or, even better, adjust this integration to use `SensorEntityDescription`s instead.

How would one do that? I'm not familiar enough with it to know how to implement it with `SensorEntityDescription`

Reply:
There are many many examples in our code base doing just that already, search for `SensorEntityDescription`.

Reply:
I think I got it.